src/bio/bio_xstream.c

```diff
@@ -979,18 +1084,14 @@ init_blobstore_ctxt(struct bio_xs_context *ctxt, int tgt_id)
 	 * Lookup device mapped to @tgt_id in the per-server metadata,
 	 * if found, create blobstore on the mapped device.
 	 */
+retry:
 	rc = smd_dev_get_by_tgt(tgt_id, &dev_info);
-	if (rc == -DER_NONEXIST) {
+	if (rc == -DER_NONEXIST && !assigned) {
 		rc = assign_device(tgt_id);
 		if (rc)
 			return rc;
-
-		rc = smd_dev_get_by_tgt(tgt_id, &dev_info);
-		if (rc) {
-			D_ERROR("Failed to get dev mapped to tgt %d. "DF_RC"\n",
-				tgt_id, DP_RC(rc));
-			return rc;
-		}
+		assigned = true;
+		goto retry;
```

Comment:
just curious is it possible to cause dead loop for any case? or maybe define a max retry times?

Reply:
It can retry only once. If assign_device() failed, it'll return error, otherwise, assigned is set to true to avoid further retry.